presto-verifier/src/main/java/com/facebook/presto/verifier/framework/LimitQueryDeterminismAnalyzer.java

```diff
@@ -135,29 +148,34 @@ private LimitQueryDeterminismAnalysis analyzeQuerySpecification(Optional<With> w
                         querySpecification.getGroupBy(),
                         querySpecification.getHaving(),
                         Optional.empty(),
-                        Optional.empty()),
+                        newLimit),
                 Optional.empty(),
                 Optional.empty());
-        return analyzeLimitNoOrderBy(queryNoLimit);
+        return analyzeLimitNoOrderBy(newLimitQuery, limit);
     }
 
-    private LimitQueryDeterminismAnalysis analyzeLimitNoOrderBy(Query queryNoLimit)
+    private LimitQueryDeterminismAnalysis analyzeLimitNoOrderBy(Query newLimitQuery, long limit)
     {
         Query rowCountQuery = simpleQuery(
                 new Select(false, ImmutableList.of(new SingleColumn(new FunctionCall(QualifiedName.of("count"), ImmutableList.of(new LongLiteral("1")))))),
-                new TableSubquery(queryNoLimit));
+                new TableSubquery(newLimitQuery));
 
         QueryResult<Long> result = callWithQueryStatsConsumer(
                 () -> prestoAction.execute(rowCountQuery, DETERMINISM_ANALYSIS, resultSet -> resultSet.getLong(1)),
                 stats -> verificationContext.setLimitQueryAnalysisQueryId(stats.getQueryId()));
 
         long rowCountNoLimit = getOnlyElement(result.getResults());
-        if (rowCountNoLimit > rowCount) {
-            return NON_DETERMINISTIC;
-        }
         if (rowCountNoLimit == rowCount) {
             return DETERMINISTIC;
         }
+        if (rowCount >= limit && rowCountNoLimit > rowCount) {
+            return NON_DETERMINISTIC;
+        }
         return FAILED_DATA_CHANGED;
     }
+
+    private static boolean isLimitAll(String limitClause)
+    {
+        return Pattern.compile("all", CASE_INSENSITIVE).matcher(limitClause).matches();
```

Comment:
nit: move Pattern.compile("all", CASE_INSENSITIVE) into a static variable to avoid re-compiling on each call

Reply:
Replaced with a simpler implementation: `limitClause.toLowerCase(ENGLISH).equals("all");`